src/data/set/lattice.lean

```diff
@@ -1377,6 +1377,127 @@ begin
+
+lemma sInter_prod_set (f₁ : set (set α)) (h₁ : f₁.nonempty) (T : set β)
+  : ⋂₀f₁ ×ˢ T  = ⋂₀((λ (C : set α), C ×ˢ T) '' (f₁)) :=
+begin
+  ext ⟨x,y⟩,
+  rw [prod_mk_mem_set_prod_eq, mem_sInter, sInter_image, mem_Inter],
+  simp only [mem_Inter, prod_mk_mem_set_prod_eq],
+  split,
+  { intros h S hS,
+    split,
+    { apply h.1, exact hS, },
+    { exact h.2, }, },
+  { intros h,
+    split,
+    { intros S hS,
+      exact mem_of_mem_inter_left (h S hS),},
+       { cases h₁ with S hS,
+      have e1: x∈ S ∧ y∈ T :=
+      begin
+        apply h,
+        exact hS,
+      end,
+      exact e1.2, }, },
+end
+
+lemma set_prod_sInter (f₂ : set (set β)) (h₁ : f₂.nonempty) (S : set α)
+  : S ×ˢ ⋂₀f₂ = ⋂₀((λ (C : set β), S ×ˢ C) '' (f₂)) :=
+begin
+  ext ⟨x,y⟩,
+  rw [prod_mk_mem_set_prod_eq, mem_sInter, sInter_image, mem_Inter],
+  simp only [mem_Inter, prod_mk_mem_set_prod_eq],
+  split,
+  { intros h S hS,
+  split,
+  { exact h.1, },
+  { apply h.2, exact hS, }, },
+  { intros h,
+    split,
+       { cases h₁ with T hT,
+      have e1: x∈ S ∧ y ∈ T :=
+      begin
+        apply h,
+        exact hT,
+      end,
+      exact e1.1, },
+    { intros S hS,
+    exact mem_of_mem_inter_right (h S hS), }, },
+end
+
+lemma sInter_prod_sInter (f₁ : set (set α)) (f₂ : set (set β)) (h₁ : f₁.nonempty) (h₂ : f₂.nonempty)
+  : ⋂₀f₁ ×ˢ ⋂₀f₂ = ⋂₀((λ (C : set α × set β), C.1 ×ˢ C.2) '' (f₁ ×ˢ f₂)) :=
+begin
+  rw subset_antisymm_iff,
+  split,
+  { apply sInter_prod_sInter_subseteq, },
+  { rintro ⟨x, y⟩,
+    intro h,
+    rw [image_prod, mem_sInter] at h,
+    rw [prod_mk_mem_set_prod_eq, mem_sInter],
+    rw nonempty_def at h₁,
+    rw nonempty_def at h₂,
+    cases h₁ with s₀ hs₀,
+    cases h₂ with t₀ ht₀,
+    split,
+    { rintro s₁ hs₁,
+      have e1 : (x,y) ∈ s₁ ×ˢ t₀ :=
+      begin
+        apply h,
+        rw mem_image2,
+        use s₁,
+        use t₀,
+        split,
+        { exact hs₁, },
+        { split,
+          { exact ht₀, },
+          { simp only [eq_self_iff_true], }, }
+      end,
+      rw prod_mk_mem_set_prod_eq at e1,
+      exact e1.1, },
+    { rintro t₁ ht₁,
+      have e1 : (x,y) ∈ s₀ ×ˢ t₁ :=
+      begin
+        apply h,
+        rw mem_image2,
+        use s₀,
+        use t₁,
+        split,
+        { exact hs₀, },
+        { split,
+          { exact ht₁, },
+          { simp only [eq_self_iff_true], }, }
+      end,
+      rw prod_mk_mem_set_prod_eq at e1,
+      exact e1.2, }, },
+end
+
+lemma sInter_prod_sInter_empty (f₁ : set (set α)) (h₁ : f₁.nonempty)
+  : ⋂₀f₁ ×ˢ ⋂₀(∅ : set (set β))  = ⋂₀((λ (C : set α), C ×ˢ univ) '' (f₁)) :=
+by rw [sInter_empty, (sInter_prod_set _ h₁)]
+
+lemma sInter__empty_prod_sInter (f₂ : set (set β)) (h₂ : f₂.nonempty)
+  : ⋂₀(∅ : set (set α)) ×ˢ ⋂₀f₂    = ⋂₀((λ (C : set β), univ ×ˢ C) '' (f₂)) :=
+by rw [sInter_empty, (set_prod_sInter _ h₂)]
+
+lemma sInter__empty_prod_sInter_empty : ⋂₀(∅ : set (set α)) ×ˢ ⋂₀(∅ : set (set β)) = univ ×ˢ univ :=
+by rw [sInter_empty, sInter_empty]
```

Comment:
These last three lemmas (but particularly this one) seem a little unnecessary.

Reply:
@tb65536 Thanks for the review. I was hoping to use these lemmas in a subsequent PR to prove something like this:
```
lemma finite_prod_open {s : set (set α)} {t : set (set β)} (f₁ : set (set α)) (f₂ : set (set β))
  [hf₁ : f₁.finite] [hf₂ : f₂.finite] (ho₁ : ∀ i ∈ f₁, (generate_from s).is_open i)
  (ho₂ : ∀ i ∈ f₂, (generate_from t).is_open i) :
  (@prod.topological_space α β (generate_from s) (generate_from t)).is_open (⋂₀f₁ ×ˢ ⋂₀f₂) :=
begin
  cases f₁.eq_empty_or_nonempty with he₁ hne₁,
  { cases f₂.eq_empty_or_nonempty with he₂ hne₂,
    { rw [he₁, he₂, sInter__empty_prod_sInter_empty, univ_prod_univ], apply is_open_univ, },
    { rw [he₁, sInter__empty_prod_sInter],
      apply is_open_sInter,
      simp only, exact finite.image univ.prod hf₂,
      intro S,
      simp only [mem_image, forall_exists_index, and_imp],
      intros x hx hS,
      rw ← hS,
      apply is_open.prod,
      apply is_open_univ,
      exact ho₂ x hx,
      exact hne₂, } },
  { cases f₂.eq_empty_or_nonempty with he₂ hne₂,
  { rw [he₂, sInter_prod_sInter_empty],
    apply is_open_sInter,
    apply finite.image,
    exact hf₁,
    intro S,
    simp only [mem_image, forall_exists_index, and_imp],
    intros x hx hS,
    rw ← hS,
    apply is_open.prod,
    exact ho₁ x hx,
    apply is_open_univ,
    exact hne₁, },
  { rw sInter_prod_sInter, apply is_open_sInter,
    simp only [image_prod], exact finite.image2 prod hf₁ hf₂,
    intro S,
    simp only [image_prod, mem_image2, exists_and_distrib_left, forall_exists_index, and_imp],
    intros x hx y hy hS,
    rw ← hS,
    apply is_open.prod,
    exact ho₁ x hx,
    exact ho₂ y hy,
    exact hne₁,
    exact hne₂, }, },
end
```

I could inline them into this lemma if you think that would be better?

Reply:
I think inlining `rw sInter_empty` would make more sense.

Reply:
@tb65536 I'm happy to do that then and have removed those lemmas from this PR. Thanks.